src/RestrictedPython/Guards.py

```diff
@@ -266,10 +266,16 @@ def safer_getattr(object, name, getattr=getattr):
     if isinstance(object, _compat.basestring) and name == 'format':
         raise NotImplementedError(
             'Using format() on a %s is not safe.' % object.__class__.__name__)
-    return getattr(object, name)
+    if name.startswith('_'):
+        raise AttributeError(
+            '"{name}" is an invalid attribute name because it '
+            'starts with "_"'.format(name=name)
+        )
+    return getattr(object, name, default)
 
 
 safe_builtins['_getattr_'] = safer_getattr
+safe_builtins['getattr'] = safer_getattr
```

Comment:
I am not sure if this line is necessary. I will play with it and we will see.

Reply:
I have tested several ways, without this line it worked again in newer Python versions (3.6+)